api/routers/edr.py

```diff
@@ -42,47 +49,41 @@ async def get_locations(
             points=[dstore.Point(lat=coord[1], lon=coord[0]) for coord in poly.exterior.coords]
         ),
     )
-    # TODO: It is not possible to request all of the locations, because the request is too large for gRPC.
-    #  Thus it is necessary to update the datastore.
 
     # TODO: Add flag to protobuf so that it only retrieves the latest observations from the datastore.
-    # TODO: Simulate retrieval of last obs by setting a recent timestamp
     ts_response = await getObsRequest(ts_request)
 
-    platform_standard_names = defaultdict(set)
-    for obs in sorted(ts_response.observations, key=lambda obs: obs.ts_mdata.platform):
-        platform_standard_names[obs.ts_mdata.platform].add(obs.ts_mdata.instrument)
+    platform_parameters: DefaultDict[str, Set[str]] = defaultdict(set)
+    platform_coordinates: Dict[str, Tuple[float, float]] = {}
+    all_parameters: Dict[str, Parameter] = {}
+    for obs in ts_response.observations:
+        parameter = Parameter(
+            description={"en": obs.ts_mdata.title},
+            observedProperty=ObservedProperty(
+                id=f"https://vocab.nerc.ac.uk/standard_name/{obs.ts_mdata.standard_name}",
+                label={"en": obs.ts_mdata.instrument},
+            ),
+            unit=Unit(label={"en": obs.ts_mdata.unit}),
+        )
+        platform_parameters[obs.ts_mdata.platform].add(obs.ts_mdata.instrument)
+        # HACK
+        platform_coordinates[obs.ts_mdata.platform] = (obs.obs_mdata[0].geo_point.lon, obs.obs_mdata[0].geo_point.lat)
```

Comment:
Why is this a hack? Because it doesn't support the case where obs location may change within a time series?

Reply:
I rewrote the code a little bit, in the new version I collect all the coordinates for a station, and check if each station has a unique location. There is a problem if different timeseries on one platform have different coordinates.

Reply:
Right, so the premise here is that this particular kind of stations will be just that: stationary? And then we need to somehow deal with mobile sensors in some other way altogether, i.e. in different part of the code?